tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpConnectionIT.java

```diff
@@ -88,36 +91,119 @@ public void testConnectSucceedsForRegisteredDevice(final String tlsVersion, fina
     }
 
     /**
-     * Verifies that the adapter opens a connection if auto-provisioning is enabled for the device certificate.
+     * Verifies that the adapter opens a connection if auto-provisioning is enabled for the device certificate, and it
+     * is not configured auto-provisioning-device-id template and auth-id template.
```

Comment:
it actually looks like you are testing that the auth-id template is being used to generate the auth-id ...

Reply:
If the auth-id template is not configured explicitly, the auth-id is generated by the subject-dn.
I could add javadoc sentence like "Verifies that the device-id is not generated by subject-dn and the auth-id is generated by the subject-dn as default template.
Would it be more clear or do you have better suggestion?

Reply:
Ok, I see your point. So your changes to this test actually makes verification of the default behavior more explicit. That's a good improvement.
IMHO it would be very helpful to have an integration test that verifies the correct application of the device-id and auth-id templates, right?

Reply:
Right, I am a little absent-minded these days. Sorry for the confusion ...